src/Blockcore.Indexer.Core/Controllers/QueryController.cs

```diff
@@ -39,6 +40,18 @@ public IActionResult GetAddress([MinLength(4)][MaxLength(100)] string address)
          return Ok(storage.AddressBalance(address));
       }
 
+      /// <summary>
+      /// Only returns addresses with outputs or spent outputs (also when balance is 0)
+      /// </summary>
+      /// <param name="addresses"></param>
+      /// <returns></returns>
+      [HttpPost]
+      [Route("addresses/balance")]
+      public IActionResult GetAddressesBalance(IList<string> addresses)
```

Comment:
I wonder if we could also make an endpoint that will take a list of addresses as url param

maybe we can do it separated with a colon 
`<address>:<address>:<address>`